### ISSUE TITLE
preload seems to have no effect on loading

### ISSUE THREAD

**Issue by matthieu-fesselier** [LDflex/LDflex#67]
On a container of 600 resources, if I measure how long it takes to loop over all elements and show 2 properties I have the following results:
```js 
await solid.data.context.extend({
  '@vocab': 'http://happy-dev.fr/owl/#',
  rdf: 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
  rdfs: 'http://www.w3.org/2000/01/rdf-schema#',
  ldp: 'http://www.w3.org/ns/ldp#',
  foaf: 'http://xmlns.com/foaf/0.1/',
  name: 'rdfs:label',
  acl: 'http://www.w3.org/ns/auth/acl#',
  permissions: 'acl:accessControl',
  mode: 'acl:mode'
});
const data = solid.data['https://api.community.hubl.world/skills/'];

// takes around 6000ms
for await (const s of data['ldp:contains']) {
  await Promise.all([s['name'].value, s['type'].value])
}

// takes around 4000ms
const skillsContainer = data['ldp:contains']
await skillsContainer.preload('rdfs:label', 'type');
for await (const s of skillsContainer) {
  await Promise.all([s['name'].value, s['type'].value]);
}
```

It seems that the preload helps but is quite slow.

However, in the second case, if I measure only the loop after a preload, it takes 200ms, which is nice!

Here is [a zip of my test case](https://github.com/LDflex/LDflex/files/5371741/ldflex-test.zip)

**Comment by balessan** [comment]
Any news on this one ?

**Comment by RubenVerborgh** [comment]
We're working on it; these are [complex times](https://www.brusselstimes.com/news/belgium-all-news/138408/consultative-committee-new-measures-alexander-de-croo-covid-19-commissioner-lockdown-frank-vandenbroucke-jan-jambon-elio-di-rupo-hospitals-limit-contacts-non-essential-shops-supermarkets-autumn-holida/) though.

**Comment by balessan** [comment]
Yes, we are also in lockdown here in France... Not an exciting time for social interactions :-)

**Comment by RubenVerborgh** [comment]
@matthieu-fesselier There's currently a missing content type on https://cdn.happy-dev.fr/owl/hdcontext.jsonld

**Comment by matthieu-fesselier** [comment]
Yes indeed, I think it's OK now

**Comment by RubenVerborgh** [comment]
The generated SPARQL query is

```sparql
SELECT ?contains ?preload_0 ?preload_1 WHERE {
  <https://api.community.hubl.world/skills/> <http://www.w3.org/ns/ldp#contains> ?contains.
  OPTIONAL {
    { ?contains <http://www.w3.org/2000/01/rdf-schema#label> ?preload_0. }
    UNION
    { ?contains <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> ?preload_1. }
  }
}
```

Test case at http://query.linkeddatafragments.org/#datasources=https%3A%2F%2Fapi.community.hubl.world%2Fskills%2F&query=SELECT%20%3Fcontains%20%3Fpreload_0%20%3Fpreload_1%20WHERE%20%7B%0A%20%20%3Chttps%3A%2F%2Fapi.community.hubl.world%2Fskills%2F%3E%20%3Chttp%3A%2F%2Fwww.w3.org%2Fns%2Fldp%23contains%3E%20%3Fcontains.%0A%20%20OPTIONAL%20%7B%0A%20%20%20%20%7B%20%3Fcontains%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23label%3E%20%3Fpreload_0.%20%7D%0A%20%20%20%20UNION%0A%20%20%20%20%7B%20%3Fcontains%20%3Chttp%3A%2F%2Fwww.w3.org%2F1999%2F02%2F22-rdf-syntax-ns%23type%3E%20%3Fpreload_1.%20%7D%0A%20%20%7D%0A%7D

Will create a Comunica issue.
